extensions/interactions/MathExpressionInput/directives/MathExpressionInput.js

```diff
@@ -45,6 +45,11 @@ oppia.directive('oppiaInteractiveMathExpressionInput', [
             EVENT_PROGRESS_NAV_SUBMITTED) {
           var guppyDivElt = $element[0].querySelector('.guppy-div');
 
+          //Dynamically assigns a unique id to the guppy-div
+          guppyDivElt.setAttribute(
+            'id', 'guppy_' + (Object.keys(Guppy.instances).length).toString());
```

Comment:
Suggest actually generating a random ID if you want it to be unique, since guppy divs get added and removed from the DOM.

Reply:
I can use number of seconds passed since epoch as ID . Is it fine?

Reply:
Still a bit fragile, I think. Why not just some version of Math.random()? (Check how we use it elsewhere; this is a problem we've tried to solve before.)

Reply:
Math.floor(Math.random()*1000) would be a safe bet.
Just one additional view :
 I checked into older Guppy version that we are using  to see how it had implented this: It was keeping track of maximum uid allocated so  far. 
`var i = Guppy.max_uid || 0;
	
	while(document.getElementById("guppy_uid_"+i)) i++;
	Guppy.max_uid = i;
	guppy_div.id = "guppy_uid_"+i;`